vladiate/validators.py

```diff
@@ -60,18 +60,23 @@ def __init__(self, **kwargs):
 class SetValidator(Validator):
     """Validates that a field is in the given set"""
 
-    def __init__(self, valid_set=[], **kwargs):
+    def __init__(self, valid_set=[], ignore_case=False, **kwargs):
         super(SetValidator, self).__init__(**kwargs)
         self.valid_set = set(valid_set)
         self.invalid_set = set([])
+        self.ignore_case = ignore_case
         if self.empty_ok:
             self.valid_set.add("")
+        if self.ignore_case:
+            self.valid_set = [s.lower() for s in valid_set]
 
     def validate(self, field, row={}):
+        if self.ignore_case:
+            field = field.lower()
         if field not in self.valid_set:
             self.invalid_set.add(field)
             raise ValidationException(
-                "'{}' is not in {}".format(field, _stringify_set(self.valid_set, 100))
+                f"'{field}' is not in {_stringify_set(self.valid_set, 100)}"
```

Comment:
IMO, this should output the original field name and set, not the lowercased field name, otherwise this might be confusing. Perhaps if `ignore_case` is true, we can append a "(ignoring case sensitivity)" to this message.

Reply:
I agree, that's something I totally overlooked - apologies!

Is splitting it into two like this viable? Performance-wise I imagine it's not the best, but could be easier for someone to debug, or is the append more worthwhile?
```python
class SetValidator(Validator):
    """Validates that a field is in the given set"""

    def __init__(self, valid_set=[], ignore_case=False, **kwargs):
        super(SetValidator, self).__init__(**kwargs)
        self.valid_set = set(valid_set)
        self.invalid_set = set([])
        self.ignore_case = ignore_case
        
        if self.empty_ok:
            self.valid_set.add("")
        if self.ignore_case:
            self.ignore_case_set = [s.lower() for s in valid_set]

    def validate(self, field, row={}):
        if self.ignore_case:
            field_ignore_case = field.lower()
            if field_ignore_case not in self.ignore_case_set:
                self.invalid_set.add(field_ignore_case)
                raise ValidationException(
                    f"'{field_ignore_case}' is not in {_stringify_set(self.ignore_case_set, 100)} (ignoring case "
                    f"sensitivity)"

                )
        else:
            if field not in self.valid_set:
                self.invalid_set.add(field)
                raise ValidationException(
                    f"'{field}' is not in {_stringify_set(self.valid_set, 100)}"
                )

    @property
    def bad(self):
        return self.invalid_set


```

Reply:
Sure, I think something like this would work too:

```
class SetValidator(Validator):
    """Validates that a field is in the given set"""

    def __init__(self, valid_set=[], ignore_case=False, **kwargs):
        super(SetValidator, self).__init__(**kwargs)
        self.valid_set = set(valid_set)
        self.invalid_set = set([])
        self.ignore_case = ignore_case
        set_to_check =  [s.lower() for s in valid_set] if self.ignore_case else valid_set
        
        if self.empty_ok:
            self.valid_set.add("")

    def validate(self, field, row={}):
        field_to_check = field.lower() if self.ignore_case else field
          if field_to_check not in self.set_to_check:
              self.invalid_set.add(field)
              raise ValidationException(
                  f"'{field}' is not in {_stringify_set(self.valid_set, 100)}"
                  + " (ignoring case sensitivity)" if self.ignore_case else ""
              )

    @property
    def bad(self):
        return self.invalid_set
```

Reply:
I'm still kind of new to Python - completely forgot you can add if/else on list comprehensions 😅. Definitely the better option here!

Pushed the change :).

Reply:
I think tests will need updated as well!